### ISSUE TITLE
Unable to Open a large CSV file 

### ISSUE THREAD

**Issue by niksingh710** [ONLYOFFICE/DesktopEditors#482]
**Do you want to request a *feature* or report a *bug*?**
Bug
**What is the current behavior?**
Shows Error when i try to open a csv file that is 242 mb
**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**
If u want i can link the file i am trying to open just opening it and selecting delimeter =comma then it loads for a while and shows the error message as in screenshot
![Uploading Screenshot_20201014_195610.png…]()

**What is the expected behavior?**
To open the file
**Which versions of DesktopEditors, and which OS are affected by this issue? Did this work in previous versions of DesktopEditors?**
this is the first time i am using only office

**Comment by Rita-Bubnova** [comment]
Hello, @niksingh710.
Can you attach CSV file?

**Comment by niksingh710** [comment]
https://www.kaggle.com/snap/amazon-fine-food-reviews

at this link, there will be a zip of 642 MB but it will be downloaded only of 242 MB after extracting u will get csv.
@Rita-Bubnova

**Comment by fruitdumonde** [comment]
I just experienced the same issue, but the file is only about 85MB (less than 600.000 lines). The attached zip file contains the csv file I was trying to open.

- ONLYOFFICE Desktop Editors version 6.0.2.10
- Windows 10 20H2 (OS Build 19042.630)
- This is the first time trying to open the file with DesktopEditors. The file loads in other applications.

[RKI_COVID19.zip](https://github.com/ONLYOFFICE/DesktopEditors/files/5610959/RKI_COVID19.zip)

**Comment by Rita-Bubnova** [comment]
@fruitdumonde, I can not reproduce your problem. I use ONLYOFFICE Desktop Editors version 6.0.2.10, Windows 10.
What delimiter do you use?

@niksingh710, sorry for late response. I can not download your file.

**Comment by fruitdumonde** [comment]
I use comma as the delimiter and UTF-8 as encoding.

In the appliaction log I found this error around the time I tried to open the file:
Faulting application name: editors_helper.exe, version: 6.0.2.10, time stamp: 0x5fadae4a
Faulting module name: ucrtbase.dll, version: 10.0.19041.546, time stamp: 0x43cbc11d
Exception code: 0xc0000409
Fault offset: 0x000000000007287e
Faulting process ID: 0x290c
Faulting application start time: 0x01d6c5a900a1fd80
Faulting application path: C:\Program Files\ONLYOFFICE\DesktopEditors\editors_helper.exe
Faulting module path: C:\WINDOWS\System32\ucrtbase.dll
Report ID: 509db7d0-5d8b-4e45-8157-e0ee7d4ccdf9
Faulting package full name: 
Faulting package-relative application ID:

**Comment by Rita-Bubnova** [comment]
I can confirm - this is bug, issue 47736 in our private issue tracker.

**Comment by niksingh710** [comment]
@Rita-Bubnova [Archive Link](https://mega.nz/file/5hJ1VAbT#ZCPeQprADeeEFmhB_G7edi9Pcwg98hSPf40xvw12yWA)
 in this zip file, there is Reviews.csv that I was using.

**Comment by Rita-Bubnova** [comment]
@niksingh710, I can confirm - this is bug, also issue 47736 in our private issue tracker.

**Comment by thibaultmol** [comment]
Issue is still relevant somehow.... This is a dealbreaker for me to use Onlyoffice. 
Libreoffice calc does it in a few seconds with no crashes. Meanwhile onlyoffice:
![image](https://user-images.githubusercontent.com/1010226/188501643-c9b79666-5ca6-4b0f-ab36-60496f0977be.png)

To test: https://eforexcel.com/wp/downloads-18-sample-csv-files-data-sets-for-testing-sales/ 
Use the '100000 Sales Records'  one

**Comment by ShockwaveNN** [comment]
@thibaultmol I've succesfully opened this file 
![image](https://user-images.githubusercontent.com/668524/188594300-44d0b3e2-dd29-4606-b404-357265875c33.png)

So we need some information about your env and DesktopEditors versions

**Comment by thibaultmol** [comment]
Onlyoffice 7.1.1.57 (Installed via the AUR. But I just uninstalled that version and installed the flatpak version to verify and that also has the same problem)


Operating System: Garuda Linux
KDE Plasma Version: 5.25.5
KDE Frameworks Version: 5.97.0
Qt Version: 5.15.5
Kernel Version: 5.19.3-zen1-1-zen (64-bit)
Graphics Platform: X11
Processors: 16 × AMD Ryzen 7 3700X 8-Core Processor
Memory: 31,3 GiB of RAM
Graphics Processor: NVIDIA GeForce GTX 1080/PCIe/SSE2

So I open onlyoffice, create a new calc (don't save it yet) then 'get data' and import that csv with the basic settings

![image](https://user-images.githubusercontent.com/1010226/188925836-a84a180e-3f33-421a-a6d5-2750e062d332.png)

**Comment by ShockwaveNN** [comment]
> So I open onlyoffice, create a new calc (don't save it yet) then 'get data' and import that csv with the basic settings

You didn't mention you've used `Get Data`, so I think this is separate issue so please file a new one

**Comment by thibaultmol** [comment]
Ah sorry yeah. Somehow realized that get data was the only way to get csv's into onlyoffice. 
Opening the CSV file does indeed work

**Comment by Juniorsnet** [comment]
I have similar issues with large files, in the example file when I try to resize all columns to automatic I got the error "Something has gone wrong"
[DATAFILE.xlsx](https://github.com/ONLYOFFICE/DesktopEditors/files/9538845/DATAFILE.xlsx)

**Comment by ShockwaveNN** [comment]
@Juniorsnet I think it's completly separate issue, your file is not csv, it's opened normally by your description and failing on resize

You should create a new issue

**Comment by Juniorsnet** [comment]
> @Juniorsnet I think it's completly separate issue, your file is not csv, it's opened normally by your description and failing on resize
> 
> You should create a new issue

Yes, is not exactly the same, but I think that the general problem is that onlyoffice can't handle large spreadsheets furthermore I have the same problem that @fruitdumonde.

I suppose that this issue is not fixed yet right?

**Comment by ShockwaveNN** [comment]
Please create a new issue and we can detect that

This issue is for csv only and you got no problem opening file, but you got problems resizing - this may be unrelated at all

**Comment by rabinnh** [comment]
Dec, 4, 2024 - Same issue here on Ubuntu 22.04, OnlyOffice 8.0.1.31 (deb).  The thing is, this comma-delimited file is only 251653 bytes, so it appears to be a coding issue somewhere.

My machine has 96GB of RAM.

File is attached.


[deb_ubuntu_vers.csv](https://github.com/user-attachments/files/18011491/deb_ubuntu_vers.csv)

**Comment by VladimirProkofiev52** [comment]
Hello! @rabinnh 
I have reviewed your issue and can confirm that it has not been resolved yet. 
Thank you for your report and the file you provided. 
Once the issue is resolved, we will update you in this thread.
